Firmware/depth-sensing-JSN-SR04_Gen3_RK-Integration-Indu

```diff
@@ -0,0 +1,326 @@
+            Serial.printf("distance error %f", result.status);
+            break;
+    }
+
+    return depth;
+}
+
+/* ---------------------- GET TEMP FUNCTION ---------------------- */
+float getTemp() {
+
+    //getting the voltage reading from the temperature sensor
+    int reading = analogRead(sensorPin);  
+
+    Serial.println(reading);
+    
+    // converting that reading to voltage, for 3.3v arduino use 3.3
+    float voltage = reading * 3.3;
+    voltage /= 4096.0; 
+    
+    // print out the voltage
+    Serial.print(voltage); Serial.println(" volts");
+    
+    // now print out the temperature
+    float temperatureC = (voltage - 0.5) * 100 ;  //converting from 10 mv per degree wit 500 mV offset
+                                                  //to degrees ((voltage - 500mV) times 100)
+    Serial.print(temperatureC); Serial.println(" degrees C");
+
+    return temperatureC;
+
+}
+
+/* ---------------------- PRINT TO SD CARD FUNCTION ---------------------- */
+void printToFile() {
+
+  if (!filenameCreated) {
+    // Get year, month, and day for filename
+    int filenum = 0; // start at zero and increment by one if file exists
+    sprintf(filename, "%02d%02d%02d%02d.csv", GPS.year, GPS.month, GPS.day, filenum);
+
+    // Check for existence of filename with current filenum
+    while (SD.exists(filename)) {
+      filenum++;
+      sprintf(filename, "%02d%02d%02d%02d.csv", GPS.year, GPS.month, GPS.day, filenum);
+    }
+    filenameCreated = true;
+  }
+
+  Serial.println(filename);
+
+  // Create filename
+  // Open the file: SPI SD comms
+  File dataFile = SD.open(filename, FILE_WRITE);
+
+  // if the file is available, write to it:
+  if (dataFile) {
+    // Date
+    dataFile.print(GPS.month, DEC);
+    dataFile.print('/');
+    dataFile.print(GPS.day, DEC);
+    dataFile.print("/20");
+    dataFile.print(GPS.year, DEC);
+    dataFile.print(",");
+
+    // Time
+    dataFile.print(GPS.hour, DEC);
+    dataFile.print(':');
+    if (GPS.minute < 10) {
+      dataFile.print('0');
+    }
+    dataFile.print(GPS.minute, DEC);
+    dataFile.print(':');
+    if (GPS.seconds < 10) {
+      dataFile.print('0');
+    }
+    dataFile.print(GPS.seconds, DEC);
+    dataFile.print(".");
+    if (GPS.milliseconds < 10) {
+      dataFile.print("00");
+    } else if (GPS.milliseconds > 9 && GPS.milliseconds < 100) {
+      dataFile.print("0");
+    }
+    dataFile.println(GPS.milliseconds);
+    dataFile.print(",");
+
+    // Elapsed Time
+    dataFile.print(millis() / 1000);
+    dataFile.print(",");
+
+    // Location
+    dataFile.print(GPS.latitude, 4);
+    dataFile.print(",");
+    dataFile.print(GPS.lat); // N or S
+    dataFile.print(",");
+    dataFile.print(GPS.longitude, 4);
+    dataFile.print(",");
+    dataFile.print(GPS.lon); // E or W
+    dataFile.print(",");
+
+    // Temperature
+    dataFile.print(temperatureC);
+    dataFile.print(",");
+
+    // Depth/distance
+    dataFile.print(depth);
+    dataFile.print(",");
```

Comment:
Here, I am writing "depth," which had received a value in the main loop, to the SD card.